### PR TITLE
Reactions

### PR DESCRIPTION
**Pull request by redsky17**

### CODE REVIEW

File: src/ChatPage.cpp [hunk 1]
```diff
@@ -606,6 +606,12 @@ ChatPage::resetUI()
         showUnreadMessageNotification(0);
 }
 
+void
+ChatPage::reactMessage(const QString &id)
+{
+        view_manager_->queueReactionMessage(current_room_, id, "👀");
+}
+
```
Comment (redsky17) [inline src/ChatPage.cpp:614]: Unused code needs to be cleaned up.

File: src/ChatPage.h [hunk 2]
```diff
@@ -87,6 +87,7 @@ class ChatPage : public QWidget
         //! Show the room/group list (if it was visible).
         void showSideBars();
         void initiateLogout();
+        void reactMessage(const QString &eventId);
```
Comment (redsky17) [inline src/ChatPage.h:90]: Unused code needs to be cleaned up.

File: src/timeline/TimelineModel.cpp [hunk 3]
```diff
@@ -941,6 +961,13 @@ TimelineModel::decryptEvent(const mtx::events::EncryptedEvent<mtx::events::msg::
         return {dummy, false};
 }
 
+void
+TimelineModel::reactAction(QString id)
+{
+        setReaction(id);
+        ChatPage::instance()->reactMessage(id);
+}
+
```
Comment (redsky17) [inline src/timeline/TimelineModel.cpp:970]: Unused code needs to be removed

File: src/timeline/TimelineModel.cpp [hunk 4]
```diff
@@ -379,7 +391,7 @@ TimelineModel::data(const QString &id, int role) const
                 else
                         return {};
         case RoomId:
-                return QVariant(QString::fromStdString(room_id(event)));
+                return QVariant(room_id_);
```
Comment (redsky17) [inline src/timeline/TimelineModel.cpp:394]: @deepbluev7 I don't remember why I made this update.  Can you verify if this makes sense, or should I change it back?

File: src/timeline/TimelineModel.h [hunk 5]
```diff
@@ -126,6 +126,8 @@ class TimelineModel : public QAbstractListModel
           int currentIndex READ currentIndex WRITE setCurrentIndex NOTIFY currentIndexChanged)
         Q_PROPERTY(std::vector<QString> typingUsers READ typingUsers WRITE updateTypingUsers NOTIFY
                      typingUsersChanged)
+        Q_PROPERTY(QString reaction READ reaction WRITE setReaction NOTIFY reactionChanged RESET
```
Comment (redsky17) [inline src/timeline/TimelineModel.h:129]: I'm pretty confident that none of these reaction additions are needed.  Should be able to remove this, the reactAction, reaction, setReaction, resetReaction, and reactionChanged functions.  I'll test when finished to make sure I don't break anything.

File: src/timeline/TimelineModel.cpp (hunk 4 above)
Comment (deepbluev7) [reply src/timeline/TimelineModel.cpp:394]: `room_id | string | Required. The ID of the room associated with this event. Will not be present on events that arrive through /sync, despite being required everywhere else.`

=> Our events may not have a room_id set

https://matrix.org/docs/spec/client_server/latest#get-matrix-client-r0-sync

File: src/timeline/TimelineModel.h (hunk 5 above)
Comment (deepbluev7) [reply src/timeline/TimelineModel.h:129]: Yeah, that even looks counterintuitive.

Comment (redsky17) [reply src/timeline/TimelineModel.h:129]: I think when I first started implementing this, I envisioned it being similar to the replies.  But then I changed my approach and didn't clean up.

File: src/timeline/TimelineModel.cpp (hunk 4 above)
Comment (redsky17) [reply src/timeline/TimelineModel.cpp:394]: Ah, gotcha, so this was part of the updates that you made.  Thanks!

**Pull request merged**